e2e/parent-assign-from-func-call/src/Loop.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+final class Loop
+{
+    /**
+     * @var LoopInterface
+     */
+    private static $instance;
+
+    public static function get()
+    {
+        if (self::$instance instanceof LoopInterface) {
+            return self::$instance;
+        }
+
+        register_shutdown_function(function () {
+            $error = error_get_last();
```

Comment:
so, the parent of the Expression:

```php
$error = error_get_last();
```

which a:

```php
register_shutdown_function(function () {
    $error = error_get_last();
}
```

is **unreachable**, because the check of : 

```php
if (self::$instance instanceof LoopInterface) {
    return self::$instance;
}
```
 
 early, which already defined in the `@var` annotation. 
 
 We need to lookup parent Node that is unreachable if Scope not found for it.

Reply:
Fixed 🎉 by recursive call `->processNodes()` with original Node of `UnreachableStatementNode` passed on `PHPStanNodeScopeResolver`:

https://github.com/rectorphp/rector-src/pull/2523/files#diff-625e519d3d496a50287cab05a2ffb79a95877d1850083f9a90fdd74958a01020